sdk/spring/azure-spring-boot-test-b2c/src/test/java/com/azure/test/b2c/selenium/B2CSeleniumITHelper.java

```diff
@@ -0,0 +1,158 @@
+                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverWin32);
+            } else if (Pattern.matches("mac.*", osName)) {
+                process = Runtime.getRuntime().exec("chmod +x " + chromedriverMac, null, dir);
+                process.waitFor();
+                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverMac);
+            } else {
+                throw new IllegalStateException("Unrecognized osName. osName = " + System.getProperty("os.name"));
+            }
+        } catch (InterruptedException | IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (process != null) {
+                process.destroy();
+            }
+        }
+    }
+
+    public B2CSeleniumITHelper(Class<?> appClass, Map<String, String> properties) throws InterruptedException {
+        emailAddress = B2C_USER_EMAIL;
+        password =  System.getenv(B2C_USER_PASSWORD);
+        app = new AppRunner(appClass);
+        DEFAULT_PROPERTIES.forEach(app::property);
+        properties.forEach(app::property);
+
+        ChromeOptions options = new ChromeOptions();
+//        options.addArguments("--headless");
+//        options.addArguments("--incognito", "--no-sandbox", "--disable-dev-shm-usage");
+        this.driver = new ChromeDriver(options);
+
+        this.app.start();
+        Thread.sleep(3000);
+    }
+
+    public void signIn(String userFlowName) throws InterruptedException {
+        driver.get(app.root());
+        Thread.sleep(3000);
+        driver.findElement(By.cssSelector("a[href='/oauth2/authorization/" + userFlowName + "']")).click();
+        Thread.sleep(3000);
+        driver.findElement(By.id("email")).sendKeys(emailAddress);
+        driver.findElement(By.id("password")).sendKeys(password);
+        driver.findElement(By.cssSelector("button[type='submit']")).click();
+        Thread.sleep(7000);
+        manualRedirection();
+    }
+
+    public void profileEditJobTitle(String newJobTitle) throws InterruptedException {
+        Thread.sleep(5000);
+        driver.findElement(By.id("profileEdit")).click();
+        Thread.sleep(3000);
+        changeJobTile(newJobTitle);
+        driver.findElement(By.cssSelector("button[type='submit']")).click();
+        Thread.sleep(5000);
+        manualRedirection();
+    }
+
+    public void logout() throws InterruptedException {
+        Thread.sleep(5000);
+        driver.findElement(By.id("logout")).click();
+        Thread.sleep(3000);
+        driver.findElement(By.cssSelector("button[type='submit']")).click();
+        Thread.sleep(3000);
+        manualRedirection();
+        driver.findElement(By.cssSelector("a[href='/oauth2/authorization/" + B2C_SIGN_UP_OR_SIGN_IN + "']")).click();
+        Thread.sleep(3000);
+        String name = driver.findElement(By.cssSelector("button[type='submit']")).getText();
+        Assert.assertEquals("Sign in",name);
```

Comment:
The assertion should be moved to the IT class, the selenium should only be responsible for interaction with the browser.

Reply:
We could provide a function called 
```
public getLoginButtonText() {
}
```

and do the verification in your test.